test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/GeneratorUtil.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.services.test.utils.core.random;
+
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * 
+ * Utility class used by the email generator
+ * 
+ * @See EmailAddressGenerator
+ *
+ */
+public class GeneratorUtil {
+
+    /**
+     * Private constructor to prevent instantiation
+     */
+    private GeneratorUtil() {}
+
+
+    /**
+     * Generate a random string from an array of characters
+     * 
+     * @param validChars to chose from
+     * @param min length of the string
+     * @param max length of the string
+     * @return
+     */
+    public static String generate(final Random random, final char[] validChars, final int min,
+                    final int max) {
+        final int size = ThreadLocalRandom.current().nextInt(min, max + 1);
+        final char[] localPart = new char[size];
+        for (int i = 0; i < size; i++) {
+            localPart[i] = validChars[random.nextInt(validChars.length - 1)];
+        }
+        return new String(localPart);
+    }
+
+    /**
+     * Choose a random position given a length
+     * 
+     * @param number
+     * @return
+     */
+    public static int chooseRandomPosition(final int length) {
+        return new Random().nextInt(length);
+    }
+
+    /**
+     * Concatenate multiple character arrays
+     * 
+     * @param tokens to concatenate
+     * @return concatenated character array
+     */
+    public static char[] concat(final char[]... tokens) {
+        int size = 0;
+        for (char[] token : tokens) {
+            size += token.length;
+        }
+        char[] concatenated = new char[size];
+        int index = 0;
+        for (char[] token : tokens) {
+            for (char c : token) {
+                concatenated[index++] = c;
+            }
+        }
+        return concatenated;
+    }
+
+    /**
+     * Check that the text contains only the valid characters mentioned
+     * 
+     * @param text to check
+     * @param valid characters
+     * @return flag indicating that the check passed
+     */
+    public static boolean checkValidityOfText(final String text, final char[] valid) {
+        final char[] local = text.toCharArray();
+        boolean flag = false;
+        for (char c : local) {
+            for (char v : valid) {
+                if (c == v) {
+                    flag = true;
+                }
+            }
+        }
+        return flag;
+    }
+
+    /**
+     * 
+     * Choose from a given length <br>
+     * If the length is n, returns approximately 1 in n as positive <br>
+     * If the length is 6, returns approximately 1 in 6 as positive <br>
+     * If the length is 2, returns approximately 1 in 2 as positive
+     * 
+     * @param length
+     * @return flag
+     */
+    public static boolean chooseRandom(final int length) {
```

Comment:
Is this a good name? Maybe something like onceOutOf(int denominator). 

Which would read 

boolean isTrue = onceOutOf(6)